### PR TITLE
[1119] emit static contact data

### PR DESCRIPTION
**Pull request by misaka**
https://trello.com/c/0HJdbyrA/1119-emit-static-contacts-data-for-ucas-api-testing

### Context

UCAS would like to have some data to illustrate how the contacts will show up in the API endpoint the will use. This adds generated contacts data that doesn't rely on us implementing the whole contacts thing.

### Changes proposed in this pull request

Change the providers serialiser to generate contacts data.

### Guidance to review

Also added some extra support for contacts and to the `apiv1 find` subcommand to `mcb`. And also raw-json support.

### CODE REVIEW

**Comment by misaka** [comment]
Example JSON output:

```
% bin/mcb apiv1 find --json 1DE
{
  "institution_code": "1DE",
  "institution_name": "Oxfordshire Teacher Training",
  "institution_type": "Y",
  "accrediting_provider": "N",
  "address1": "Oxfordshire Teacher Training",
  "address2": "The Cherwell School North Site",
  "address3": "Marston Ferry Road",
  "address4": "Oxford",
  "postcode": "OX2 7EE",
  "region_code": null,
  "scheme_member": "Y",
  "contact_name": "Oxfordshire Teacher Training",
  "email": "info@ott-scitt.org.uk",
  "telephone": "01865 518209",
  "recruitment_cycle": "2019",
  "utt_application_alerts": "Yes - only my programmes",
  "type_of_gt12": "No response",
  "contacts": [
    {
      "type": "admin_contact",
      "name": "Admin Contact 1DE",
      "email": "admin_contact@ott-scitt.org.uk",
      "telephone": "01865 518209"
    },
    {
      "type": "utt_correspondent",
      "name": "Utt Correspondent 1DE",
      "email": "admin_contact@ott-scitt.org.uk",
      "telephone": "01865 518209"
    },
    {
      "type": "web_link_correspondent",
      "name": "Web Link Correspondent 1DE",
      "email": "admin_contact@ott-scitt.org.uk",
      "telephone": "01865 518209"
    },
    {
      "type": "fraud_contact",
      "name": "Fraud Contact 1DE",
      "email": "admin_contact@ott-scitt.org.uk",
      "telephone": "01865 518209"
    },
    {
      "type": "finance_contact",
      "name": "Finance Contact 1DE",
      "email": "admin_contact@ott-scitt.org.uk",
      "telephone": "01865 518209"
    },
    {
      "type": "application_alert_recipient",
      "name": "Application Alert Recipient 1DE",
      "email": "admin_contact@ott-scitt.org.uk",
      "telephone": "01865 518209"
    }
  ],
  "campuses": [
    {
      "campus_code": "S",
      "name": "Oxfordshire Teacher Training South",
      "region_code": "02"
    },
    {
      "campus_code": "C",
      "name": "Oxfordshire Teacher Training Central",
      "region_code": "02"
    },
    {
      "campus_code": "N",
      "name": "Oxfordshire Teacher Training North",
      "region_code": "02"
    },
    {
      "campus_code": "2",
      "name": "Southern Hub",
      "region_code": "02"
    },
    {
      "campus_code": "1",
      "name": "The Cherwell OTSA SCITT",
      "region_code": "02"
    }
  ]
}
```

**Comment by misaka** [comment]
Example mcb output:

```
% bin/mcb apiv1 find 1DE
+------------------------+--------------------------------+
| institution_code       | 1DE                            |
| institution_name       | Oxfordshire Teacher Training   |
| institution_type       | Y                              |
| accrediting_provider   | N                              |
| address1               | Oxfordshire Teacher Training   |
| address2               | The Cherwell School North Site |
| address3               | Marston Ferry Road             |
| address4               | Oxford                         |
| postcode               | OX2 7EE                        |
| region_code            |                                |
| scheme_member          | Y                              |
| contact_name           | Oxfordshire Teacher Training   |
| email                  | info@ott-scitt.org.uk          |
| telephone              | 01865 518209                   |
| recruitment_cycle      | 2019                           |
| utt_application_alerts | Yes - only my programmes       |
| type_of_gt12           | No response                    |
+------------------------+--------------------------------+

Campuses:
campus_code | name                           | region_code
------------|--------------------------------|------------
S           | Oxfordshire Teacher Trainin... | 02
C           | Oxfordshire Teacher Trainin... | 02
N           | Oxfordshire Teacher Trainin... | 02
2           | Southern Hub                   | 02
1           | The Cherwell OTSA SCITT        | 02

Contacts:
type                        | name                           | email                          | telephone
----------------------------|--------------------------------|--------------------------------|-------------
admin_contact               | Admin Contact 1DE              | admin_contact@ott-scitt.org.uk | 01865 518209
utt_correspondent           | Utt Correspondent 1DE          | utt_correspondent@ott-scitt... | 01865 518209
web_link_correspondent      | Web Link Correspondent 1DE     | web_link_correspondent@ott-... | 01865 518209
fraud_contact               | Fraud Contact 1DE              | fraud_contact@ott-scitt.org.uk | 01865 518209
finance_contact             | Finance Contact 1DE            | finance_contact@ott-scitt.o... | 01865 518209
application_alert_recipient | Application Alert Recipient... | application_alert_recipient... | 01865 518209
```

File: spec/requests/api/v1/provider_spec.rb [hunk 1]
```diff
@@ -202,6 +278,44 @@ def get_provider_codes_from_body(body)
                                   'recruitment_cycle' => '2019',
                                   'type_of_gt12' => 'Not coming',
                                   'utt_application_alerts' => 'Yes - for accredited programmes only',
+                                  'contacts' => [
+                                    {
+                                      "type" => "admin_contact",
```
Comment (dankmitchell) [inline spec/requests/api/v1/provider_spec.rb]: Do we need `__contact` given that we're already in the context on "contacts"

**Comment by dankmitchell** [comment]
Looks good @misaka, just one minor comment. I think @benilovj is best placed to approve 👌

**Comment by misaka** [comment]
> Looks good @misaka, just one minor comment. I think @benilovj is best placed to approve 👌

I've removed `*_contact` and `*_correspondent` from the type names.

**Comment by misaka** [comment]
[Emit static contacts data for UCAS API testing](https://trello.com/c/0HJdbyrA/1119-emit-static-contacts-data-for-ucas-api-testing)

**Pull request merged**